src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbBuffer.cs

```diff
@@ -14,21 +14,14 @@ internal abstract class DbBuffer : SafeHandle
     {
         private readonly int _bufferLength;
 
-        protected DbBuffer(int initialSize) : base(IntPtr.Zero, true)
+        protected unsafe DbBuffer(int initialSize) : base(IntPtr.Zero, true)
         {
+            Debug.Assert(initialSize % IntPtr.Size == 0, $"Expected aligned {nameof(initialSize)}.");
```

Comment:
What guarantees this?

Reply:
`DbBuffer` is is only instantiated from the derived class `CNativeBuffer`.

The buffer size is either 4096 bytes, or a larger value calculated in `CalcParameterBufferSize`.

https://github.com/dotnet/runtime/blob/b593c9beab5972fdee84c21bb641e81e5a32db36/src/libraries/System.Data.Odbc/src/System/Data/Odbc/OdbcParameterCollection.cs#L112-L128

Reply:
Perhaps it makes sense to calculate aligned buffer size in the ctor instead.